packages/components/src/combobox-control/stories/index.js

```diff
@@ -287,3 +287,39 @@ _default.args = {
 	__next36pxDefaultSize: false,
 	allowReset: false,
 };
+
+const NormalizedHyphensComboboxControl = () => {
+	const [ value, setValue ] = useState( null );
```

Comment:
Our Storybook is still quite a mixed bag, but we are moving in a direction where we try to highlight practical usage examples, rather than treating it as a place to keep testing fixtures. And as much as I like this story, I feel like this normalization feature is a bit too specific to dedicate a story to. How about we move it to a unit test, instead of a story? Doesn't have to be comprehensive, just some assertion of the expected normalization behavior like:

```js
it( 'should normalize hyphen-like characters to hyphens', () => {
  expect( normalizeTextString( 'foo~bar' ).toBe( 'foo-bar' );
};
```

I wish the `matchingSuggestions` would be refactored to a pure function so that whole logic could be unit tested, but that's for another occasion 😄